app/Models/User.php

```diff
@@ -795,6 +833,18 @@ public function verify($semester): User
         return $this;
     }
 
+    /**
+     * Sets the tenant_until date to the end of the current semester if the user is an active collegist with the tenant role.
+     * Added three months and two weeks to the end of the semester to allow the activation to happen.
+     */
+    public function setTenantDateForTenantCollegists(): bool
+    {
+        if ($this->isTenant() && $this->isActive()) {
+            return $this->personalInformation()->update(['tenant_until'=>Semester::current()->getEndDate()->addMonths(3)->addWeeks(2)]);
```

Comment:
https://github.com/EotvosCollegium/mars/issues/144 a solution can be to set the date to next semester's start